messaging/messaging_test.go

```diff
@@ -1126,6 +1126,43 @@ func TestSend(t *testing.T) {
 	}
 }
 
+func TestSendWithCustomEndpoint(t *testing.T) {
+	var tr *http.Request
+	var b []byte
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		tr = r
+		b, _ = ioutil.ReadAll(r.Body)
+		w.Header().Set("Content-Type", "application/json")
+		w.Write([]byte("{ \"name\":\"" + testMessageID + "\" }"))
+	}))
+	defer ts.Close()
+
+	ctx := context.Background()
+
+	conf := *testMessagingConfig
+	optEndpoint := option.WithEndpoint(ts.URL)
+	conf.Opts = append(conf.Opts, optEndpoint)
```

Comment:
No need to append here. Only `WithEndpoint()` should be enough:

```
conf.Opts = []option.ClientOption{
    option.WithEndpoint(ts.URL),
}
```

Reply:
It gives an error without append:
```go
conf := *testMessagingConfig
conf.Opts = []option.ClientOption{
    option.WithEndpoint(ts.URL),
}

client, err := NewClient(ctx, &conf)
if err != nil {
    t.Fatal(err)
}
```
```
--- FAIL: TestSendWithCustomEndpoint (0.34s)
    messaging_test.go:1149: google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
```

Reply:
Looks like the transport is still looking for credentials, which is ok. We can leave this as is.